alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -156,5 +162,49 @@ class AlertingSettings {
             -1L,
             Setting.Property.NodeScope, Setting.Property.Dynamic
         )
+
+        val TOTAL_MAX_ACTIONS_PER_TRIGGER = Setting.intSetting(
+            "plugins.alerting.max_actions_per_trigger",
+            DEFAULT_TOTAL_MAX_ACTIONS_PER_TRIGGER,
+            -1, MaxActionsPerTriggersValidator(),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+        val TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS = Setting.intSetting(
+            "plugins.alerting.total_max_actions_across_triggers",
+            DEFAULT_TOTAL_MAX_ACTIONS_ACROSS_TRIGGERS,
+            -1, TotalMaxActionsAcrossTriggersValidator(),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        internal class TotalMaxActionsAcrossTriggersValidator : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {}
+
+            override fun settings(): MutableIterator<Setting<*>> {
+                val settings = mutableListOf<Setting<*>>(
+                    TOTAL_MAX_ACTIONS_PER_TRIGGER
+                )
+                return settings.iterator()
+            }
+        }
+
+        internal class MaxActionsPerTriggersValidator : Setting.Validator<Int> {
+            override fun validate(value: Int) {}
+
+            override fun validate(value: Int, settings: Map<Setting<*>, Any>) {
+                if (value > MonitorRunnerService.monitorCtx.totalMaxActionsAcrossTriggers)
+                    logger.warn(
+                        "Updating maximum amount of actions per trigger to a bigger value than the maximum allowed actions across triggers."
+                    )
```

Comment:
Is it possible to prevent it from updating the setting?

Reply:
> Is it possible to prevent it from updating the setting?

I think we can leave it like that. That was what we decided to go in favor of not stopping the updates but sending warnings when they were exceeded I believe?

Reply:
I think we should stop the setting from being updated since this would break functionality for them and then we can throw an error stating that it is not allowed

Reply:
understood. stopped the setting from being updated